### PR TITLE
feat: Create Algolia PR environments

### PR DESCRIPTION
**Pull request by peterstarling**
* Creates a CI job to store Algolia index dump from dev index
* Sets up a new Algolia index in Algolia dev account on every new PR raised, then tears it down
* Updates the env var setup for PR/Dev/Prod for Algolia App ID, sets up a new Algolia index var

see: https://trello.com/c/BvZW2XBK/1342-create-pr-environments

### CODE REVIEW

**Comment by peterstarling** [comment]
@kmaid 
image
i think this setup works, at least for the PR env it does

File: apps/asap-server/scripts/restore-algolia-index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  const stream = fs
+    .createReadStream(process.env.ALGOLIA_BACKUP)
+    .pipe(StreamArray.withParser());
+  let chunks = [];
+
+  await stream
+    .on('data', ({ value }) => {
+      chunks.push(value);
+      if (chunks.length === 10000) {
```
Comment (harlan-kohll) [inline apps/asap-server/scripts/restore-algolia-index.js:19]: Where does this number come from?

File: apps/asap-server/scripts/create-algolia-backup.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const algoliasearch = require('algoliasearch');
+const fs = require('fs/promises');
+
+const main = async () => {
+  console.log({
+    appId: process.env.ALGOLIA_APP_ID,
+    apiKey: process.env.ALGOLIA_API_KEY,
```
Comment (kmaid) [inline apps/asap-server/scripts/create-algolia-backup.js:7]: Should we be revealing secrets in logs?

Comment (kmaid) [reply apps/asap-server/scripts/create-algolia-backup.js:7]: probably should also include ALGOLIA_INDEX

File: apps/asap-server/scripts/create-algolia-backup.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  });
+
+  const index = algoliasearch(
+    process.env.ALGOLIA_APP_ID,
+    process.env.ALGOLIA_API_KEY,
+  ).initIndex(process.env.ALGOLIA_INDEX);
+
+  let hits = [];
+
+  await index.browseObjects({
```
Comment (kmaid) [inline apps/asap-server/scripts/create-algolia-backup.js:17]: Won't this consume a lot of memory? What happened to the copy index function?

**Review by kmaid — CHANGES_REQUESTED** [review]
Could these files also be typescript and run through TS-Node

File: apps/asap-server/scripts/restore-algolia-index.js (hunk 1 above)
Comment (peterstarling) [reply apps/asap-server/scripts/restore-algolia-index.js:19]: oh its an arbitrary number as per algolia guidebook: https://www.algolia.com/doc/guides/sending-and-managing-data/send-and-update-your-data/how-to/sending-records-in-batches/#using-the-api

File: apps/asap-server/scripts/create-algolia-backup.js (hunk 2 above)
Comment (peterstarling) [reply apps/asap-server/scripts/create-algolia-backup.js:7]: we should not reveal them, no.

As for the TS - seems like an unnecessary effort. There is algolia CLI which technically enables us to do all that (as in exports and imports) but i kinda wanted to deliver a more custom solution, not sure if that was the right choice. Whats your opinion on that @harlan-kohll @kmaid

File: apps/asap-server/scripts/create-algolia-backup.js (hunk 3 above)
Comment (peterstarling) [reply apps/asap-server/scripts/create-algolia-backup.js:17]: well copying index is copying index and this script serves a different purpose which is to create a db dump into a file

Comment (peterstarling) [reply apps/asap-server/scripts/create-algolia-backup.js:12]: but to answer your other question - it probably will, i could batch it and use streams but for that we'd need a writeable object/array type of stream and it just seemed a little unnecessary given that we dont really have that many research outputs. Do you think i should look into it anyway?

File: apps/asap-server/scripts/create-algolia-backup.js (hunk 2 above)
Comment (kmaid) [reply apps/asap-server/scripts/create-algolia-backup.js]: If it covers our usecase then yes I think we should. The less code we write the less we have to maintain

File: apps/asap-server/scripts/create-algolia-backup.js (hunk 3 above)
Comment (kmaid) [reply apps/asap-server/scripts/create-algolia-backup.js:12]: I think it would be better for it not to fail at somepoint in the future

**Review by kmaid — APPROVED** [review]

**Pull request merged**